### PR TITLE
add the "M" to the end of the reserve-memory flag

### PR DESCRIPTION
**Pull request by joejulian**
In theory, this should be a no-op. Mostly this is for consistency and disambiguation, to make the command line easier for humans to interpret.

### CODE REVIEW

File: charts/redpanda/templates/statefulset.yaml [hunk 1]
```diff
@@ -275,7 +275,7 @@ spec:
             - start
             - --smp={{ .Values.resources.cpu.cores }}
             - --memory={{ template "redpanda-memory" . }}M
-            - --reserve-memory={{ template "redpanda-reserve-memory" . }}
+            - --reserve-memory={{ template "redpanda-reserve-memory" . }}M
```
Comment (alejandroEsc) [inline charts/redpanda/templates/statefulset.yaml:278]: The values file describe this unit, and should it always be an M?

**Review by alejandroEsc — COMMENTED** [review]
Only one question.

File: charts/redpanda/templates/statefulset.yaml (hunk 1 above)
Comment (joejulian) [reply charts/redpanda/templates/statefulset.yaml:278]: Yes, we convert it in the template from SI units to a int64 (in MiB)

**Review by alejandroEsc — APPROVED** [review]
lgtm, we can discuss tomorrow.

**Review by RafalKorepta — APPROVED** [review]

**Pull request merged**